runtime/common/src/lib.rs

```diff
@@ -484,6 +486,80 @@ pub mod foreign_investments {
 			}
 		}
 	}
+
+	/// Provides means of applying the decimals of an incoming currency to the
+	/// amount of an outgoing currency.
+	///
+	/// NOTE: Either the incoming (in case of `ConversionFromAssetBalance`) or
+	/// outgoing currency (in case of `ConversionToAssetBalance`) is assumed
+	/// to be `CurrencyId::Native`.
+	pub struct NativeBalanceDecimalConverter<AssetRegistry>(PhantomData<AssetRegistry>);
+
+	impl<AssetRegistry> ConversionToAssetBalance<Balance, CurrencyId, Balance>
+		for NativeBalanceDecimalConverter<AssetRegistry>
+	where
+		AssetRegistry: Inspect<
+			AssetId = CurrencyId,
+			Balance = Balance,
+			CustomMetadata = cfg_types::tokens::CustomMetadata,
+		>,
+	{
+		type Error = DispatchError;
+
+		fn to_asset_balance(
+			balance: Balance,
+			currency_in: CurrencyId,
+		) -> Result<Balance, DispatchError> {
+			match currency_in {
+				CurrencyId::Native => Ok(balance),
+				CurrencyId::ForeignAsset(_) => {
+					let to_decimals = AssetRegistry::metadata(&currency_in)
+						.ok_or(DispatchError::CannotLookup)?
+						.decimals;
+					convert_balance_decimals(
+						cfg_primitives::currency_decimals::NATIVE,
+						to_decimals,
+						balance,
+					)
+					.map_err(DispatchError::from)
+				}
+				_ => Err(DispatchError::Token(sp_runtime::TokenError::Unsupported)),
+			}
+		}
```

Comment:
Question 1: as I read this method, it seems like I want to transform a `currency_in` into an asset balance, but the implementation seems to do the opposite, transforming an asset balance to a native one. (Sure, I'm missing something). Is `currency_in` actually `currency_out`?

Question 2: Does it make sense to call this method with a native balance? Or should instead be an error?

Reply:
1. I followed the naming convention from the orderbook and foreign investments such that `currency_in` refers to the _incoming_ asset for which we want the balance translation. I agree it is not optimal, probably `currency_to` much better. We could also just remove the direction and call it `currency_id`. WDYTH?

2. Since calling it with native balance does not introduce any overhead, I think it should not error out because that makes it more flexible. However, I do not have a strong opinion here.

Reply:
1. Probably it's me that I feel dyslexic as with right or left 😆. I think I asked about this like 10 times haha. Given that the convention in other places is the same, we can leave it as it is. `currency_id` also fits 👍🏻 

Ok, if semantically makes sense that I could call `to_asset_balance()` using a native balance expecting no change on it, then I'm fine with the current implementation.